pkg/adapter/adapter.go

```diff
@@ -302,56 +305,127 @@ func (a *Adapter) PollForMessages(channel *wabbit.Channel,
 	}
 }
 
-func (a *Adapter) postMessage(msg wabbit.Delivery) error {
-	a.logger.Info("url ->" + a.httpMessageSender.Target)
-	req, err := a.httpMessageSender.NewCloudEventRequest(a.context)
-	if err != nil {
-		return err
+func msgContentType(msg wabbit.Delivery) (string, error) {
+	var contentType string
+
+	for key, val := range msg.Headers() {
+		if strings.ToLower(key) == "content-type" {
+			contentType = strings.ToLower(fmt.Sprint(val))
+			break
+		}
+	}
+
+	if contentType == "" {
+		return "", fmt.Errorf("no content type present on Rabbitmq msg")
 	}
 
-	// TODO: See if the message is already a Cloud Event and if so, do not wrap, just use as is.
+	switch contentType {
+	case cloudevent.ApplicationCloudEventsJSON, cloudevent.ApplicationCloudEventsBatchJSON:
+		return contentType, nil
+	default:
+		return cloudevent.ApplicationJSON, nil
+	}
+}
+
+func setEventContent(a *Adapter, msg wabbit.Delivery, contentType string) (cloudevent.Event, error) {
 	event := cloudevents.NewEvent()
-	if msg.MessageId() != "" {
-		event.SetID(msg.MessageId())
+	if contentType == cloudevent.ApplicationCloudEventsJSON {
+		err := json.Unmarshal(msg.Body(), &event)
+		if err != nil {
+			return event, err
+		}
 	} else {
-		event.SetID(string(uuid.NewUUID()))
+		if msg.MessageId() != "" {
+			event.SetID(msg.MessageId())
+		} else {
+			event.SetID(string(uuid.NewUUID()))
+		}
+
+		event.SetTime(msg.Timestamp())
+		event.SetType(sourcesv1alpha1.RabbitmqEventType)
+		event.SetSource(sourcesv1alpha1.RabbitmqEventSource(
+			a.config.Namespace,
+			a.config.Name,
+			a.config.Topic,
+		))
+		event.SetSubject(msg.MessageId())
+		event.SetExtension("key", msg.MessageId())
+
+		err := event.SetData(contentType, msg.Body())
+		if err != nil {
+			return event, err
+		}
 	}
-	event.SetTime(msg.Timestamp())
-	event.SetType(sourcesv1alpha1.RabbitmqEventType)
-	event.SetSource(sourcesv1alpha1.RabbitmqEventSource(a.config.Namespace, a.config.Name, a.config.Topic))
-	event.SetSubject(msg.MessageId())
-	event.SetExtension("key", msg.MessageId())
-
-	// TODO: Check the content type and use it instead.
-	err = event.SetData(*cloudevents.StringOfApplicationJSON(), msg.Body())
+
+	return event, nil
+}
+
+func setEventBatchContent(a *Adapter, msg wabbit.Delivery) ([]cloudevent.Event, error) {
+	// Would be good to have a batch size parameter
+	// Process all the events, so it does not send any events if any errors occurs
+	var payload []cloudevent.Event
+	err := json.Unmarshal(msg.Body(), &payload)
 	if err != nil {
-		return err
+		return payload, err
 	}
 
-	err = http.WriteRequest(a.context, binding.ToMessage(&event), req)
+	return payload, nil
+}
+
+func (a *Adapter) postMessage(msg wabbit.Delivery) error {
+	a.logger.Info("url ->" + a.httpMessageSender.Target)
+	req, err := a.httpMessageSender.NewCloudEventRequest(a.context)
 	if err != nil {
 		return err
 	}
 
-	res, err := a.httpMessageSender.Send(req)
+	a.logger.Info(fmt.Sprint(msg.Headers()))
```

Comment:
probably do not want to log all the headers in case they contain something sensitive?

Reply:
that was just me testing things and forgetting to delete them hehe :p